src/main/java/org/threadly/litesockets/tcp/SSLClient.java

```diff
@@ -184,15 +184,15 @@ private void doHandShake() throws IOException {
         select.select(100);
       }
       channel.register(select, 0);
-
-      while(hs == NEED_UNWRAP ) {
```

Comment:
These while statements where the problem, they didnt allow us to run the bigger loop which is where we check the timeout.  This might make normal ssl connections a little slower but they will use less cpu.